orchestrator.py

```diff
@@ -101,16 +101,26 @@ def toggle_logging(self, _) -> None:
             topic = module.stop if self.currently_logging else module.start
             self.mqtt_client.publish(str(topic))
 
-        self.mqtt_client.publish(
-            str(
-                topics.BOOST.stop
-                if self.currently_logging
-                else topics.BOOST.start
-            )
-        )
-
-        # `self.currently_logging` will be updated when we receive the message
-        # we publish above.
+        # Publish to V3 start topic if button on display is pressed
+        # If we are currently logging -> false payload
+        if self.currently_logging:
+            msg = dumps(
+                {"start": False}
+            )  # JSON formatted string, this is the payload
+            self.mqtt_client.publish(topics.V3.start, msg)  # Publish
+            self.set_logging_state(False)  # Changing the logging state
+            print(
+                "Stopping Logging"
+            )  # Check that at least if statement is executing
+
+        # If not -> true payload
+        else:
+            msg = dumps({"start": True})
+            self.mqtt_client.publish(topics.V3.start, msg)  # Publish
+            self.set_logging_state(True)  # Changing the logging state
+            print(
+                "Starting Logging"
+            )  # Check that at least if statement is executing
```

Comment:
Works great! (haven't tested again but witnessed it on the primary display yesterday)

Although not a huge issue, there's a bit of code duplication here. Consider something similar to the following:
```python
next_logging_state = not self.currently_logging
msg = dumps({"start": next_logging_state})
self.mqtt_client.publish(topics.V3.start, msg)
self.set_logging_state(next_logging_state)
print(f"Set logging state to {next_logging_state}")
```

Reply:
Definitely a much more efficient and concise alternative to the code I've written, I've rewritten the code to reflect this suggestion.